app/views/play/level/tome/SpellView.coffee

```diff
@@ -694,6 +711,30 @@ module.exports = class SpellView extends CocoView
     return if @destroyed
     Backbone.Mediator.publish 'tome:hide-problem-alert', {}
 
+  putSpade: () ->
+    # This happens when 1) The user has taken longer than 10 minutes, 2) the Contact button is pressed.
+    # Let's check to make sure SPADE exists.
+    if @spade
+      console.log('SPADE saving in view')
+      # Compiles the raw events (removes duplicates/0-references the timestamps)
+      spadeEvents = @spade.compile()
+      # Condenses the events further from an object with objects to a flat array.
+      condensedEvents = @spade.compress(spadeEvents)
+      # No need to write if there was absolutely nothing recorded.
+      if(condensedEvents.length is 0)
+        console.warn('SPADE saw no events, nothing to record')
+        return
```

Comment:
We don't need a comment, a log, and a self-explanatory if-statement all saying the same thing. Also, don't check against 0 and don't put parens in CoffeeScript conditionals. We can also return if spadeEvents is empty instead of trying to condense first. What about just–

`return unless spadeEvents.length`

Also, the since we are calling these `condensedEvents`, compress/uncompress in Spade itself should be called condense/expand or something, and the following `compressedEvents` logic should probably move into Spade itself.